tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is an end-to-end test. Test creates volumes of different format-types
+// and verifies their properties from ESX as well as docker host.
+// Properties being verified - capacity, disk-format and vm-attached field.
+
+// Test assumes that SSH cert has been setup to enable password-less login to VM and ESX.
+
+package e2e
+
+import (
+	"log"
+	"reflect"
+	"strings"
+
+	dockerconst "github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testName           = "volumeproperty"
+	diskStatusDetached = "detached"
+	size               = "100MB"
+	diskFormatOption   = " -o diskformat="
+)
+
+type VolumePropertyTestSuite struct {
+	volumeNames           []string
+	containerName         string
+	formatTypes           []string
+	dockerVolumeRemoveCmd string
+	volumeAttached        bool
+	config                *inputparams.TestConfig
+}
+
+func (s *VolumePropertyTestSuite) SetUpTest(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping volumeproperty_test.TestVolumeProperties.")
+	}
+	s.containerName = inputparams.GetContainerNameWithTimeStamp(testName)
+	s.formatTypes = []string{"thin", "zeroedthick", "eagerzeroedthick"}
+	s.dockerVolumeRemoveCmd = dockerconst.RemoveVolume
+	log.Println(" SETUP Finished: volumeproperty_test.TestVolumeProperties")
```

Comment:
This should be replaced with something similar to https://github.com/vmware/docker-volume-vsphere/blob/master/tests/utils/misc/misc.go#L53 .. you may introduce new log util.